cmake/MiscCheck.cmake

```diff
@@ -306,8 +306,9 @@ endif()
 # Also, we will turn off deprecated-declarations
 # due to protobuf.
 
-if(IOS)
+if(IOS AND (${IOS_ARCH} MATCHES "armv7*"))
   add_definitions("-mfpu=neon-fp16")
+  add_definitions("-arch" ${IOS_ARCH})
```

Comment:
I found that XNNPACK has more comprehensive flag configuration https://github.com/google/XNNPACK/blob/02bb42942798fdd0cf444b0fc90b6e20b847688f/CMakeLists.txt#L2729. Not sure if we want to use this, too.

Reply:
To explain this change. `-mfpu=neon-fp16` is only supported on armv7 devices (iPhone 4, 4s, iPhone 5, 5c, 5s). Nowadays we would normally target arm64 which `-march=armv8.2-a+fp16` and related should be used.